client-java/src/main/java/uk/gov/gchq/palisade/client/java/JavaClient.java

```diff
@@ -0,0 +1,233 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.java;
+
+import io.micronaut.runtime.event.annotation.EventListener;
+import org.slf4j.*;
+
+import uk.gov.gchq.palisade.client.java.download.*;
+import uk.gov.gchq.palisade.client.java.job.*;
+import uk.gov.gchq.palisade.client.java.request.*;
+import uk.gov.gchq.palisade.client.java.resource.*;
+import uk.gov.gchq.palisade.client.java.util.Bus;
+
+import javax.inject.Singleton;
+import javax.websocket.ContainerProvider;
+
+import java.net.URI;
+import java.util.*;
+import java.util.function.UnaryOperator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import static uk.gov.gchq.palisade.client.java.job.IJobContext.createJobContext;
+
+/**
+ * <p>
+ * The main client implementation.
+ * </p>
+ * <p>
+ * Note that this class should only be created via the static {@code create}
+ * methods on the {@link Client} interface. For tests it can be injected into a
+ * test class annotated with {@code @MicronautTest} as it is configured for
+ * dependency injection. This makes testing much easier.
+ *
+ * @author dbell
+ * @since 0.5.0
+ */
+@Singleton
+public class JavaClient implements Client {
+
+    private static final Logger log = LoggerFactory.getLogger(JavaClient.class);
+    private static final String EVENT_CAUGHT = "Caught event: {}";
+
+    private final ClientContext clientContext;
+    private final Map<String, JobContext> jobs = new HashMap<>();
+
+    /**
+     * Create a new JavaClient injected with required services. The main injection
+     * is the actual ApplicationContext that is performing the injection. Same as
+     * Guice, Micronaut can inject itself, which is cool.
+     *
+     * @param clientCcontext  The {@code ApplicationContext} doing the injecting
+     */
+    public JavaClient(ClientContext clientCcontext) {
+        this.clientContext = Objects.requireNonNull(clientCcontext);
+    }
+
+    @Override
+    public Result submit(UnaryOperator<JobConfig.Builder> func) {
+        return submit(func.apply(JobConfig.builder()).build());
+    }
+
+    @Override
+    public Result submit(JobConfig jobConfig) {
+
+        log.debug("Job configuration submitted: {}", jobConfig);
+
+        var palisadeRequest = createRequest(jobConfig); // actual instance sent to palisade Service
+        var palisadeService = clientContext.get(PalisadeClient.class); // the actual http client is wrapped
+
+        log.debug("Submitting request to Palisade....");
+
+        var palisadeResponse = palisadeService.submit(palisadeRequest);
+
+        assert palisadeResponse != null : "No response back from palisade service";
```

Comment:
Handle properly rather than throwing AssertionError, suggest Optional.ofNullable in-line with other code

Reply:
>  They should never be triggered in production code, and are indicative of a bug or misuse of a code path. They can be activated at run-time by way of the -ea option on the java command, but are not turned on by default.

Reply:
Running tests will automatically enable the `-ea` option. Asserts are useful on package/private methods. On public/protected it's best to check using normal methods.